middleware/test/helpers.go

```diff
@@ -279,6 +279,25 @@ func NextHandler(rcode int, err error) Handler {
 	})
 }
 
+// MockHandler returns a Handler that returns an answer for the quesion in the
+// request per the question->answer map qMap.
+func MockHandler(qMap map[dns.Question]dns.Msg) Handler {
```

Comment:
Please move this to `kubernetes/handler_test.go` and call it "autoPathHandler" or "testHandler"?